test/doctests.jl

```diff
@@ -0,0 +1,165 @@
+module DocTestsTest
+using Documenter, Compat.Test
+using Compat: @info
+
+const bad_mdstr = """
+```@docs
+DocTestsTest.Foo.foo
+```
+```jldoctest
+julia> 1 + 2
+2
+
+julia> 1 + 1
+2
+```
+```jldoctest
+julia> 1 + 1
+2
+
+julia> 1 + 2
+2
+```
+```jldoctest
+julia> 1 + 2
+2
+
+julia> 1 + 2
+2
+```
+```jldoctest
+1 + 2
+
+# output
+
+2
+```
+"""
+
+const good_mdstr = """
+```@docs
+DocTestsTest.Foo.foo
+```
+```jldoctest
+julia> 1 + 2
+3
+
+julia> 1 + 1
+2
+```
+```jldoctest
+julia> 1 + 1
+2
+
+julia> 1 + 2
+3
+```
+```jldoctest
+julia> 1 + 2
+3
+
+julia> 1 + 2
+3
+```
+```jldoctest
+1 + 2
+
+# output
+
+3
+```
+"""
+
+const bad_jlstr = """
+module Foo
+\"\"\"
+```jldoctest
+julia> 1 + 2
+2
+
+julia> 1 + 1
+2
+```
+```jldoctest
+julia> 1 + 1
+2
+
+julia> 1 + 2
+2
+```
+```jldoctest
+julia> 1 + 2
+2
+
+julia> 1 + 2
+2
+```
+```jldoctest
+1 + 2
+
+# output
+
+2
+```
+\"\"\"
+foo() = 1
+end # module
+"""
+
+const good_jlstr = """
+module Foo
+\"\"\"
+```jldoctest
+julia> 1 + 2
+3
+
+julia> 1 + 1
+2
+```
+```jldoctest
+julia> 1 + 1
+2
+
+julia> 1 + 2
+3
+```
+```jldoctest
+julia> 1 + 2
+3
+
+julia> 1 + 2
+3
+```
+```jldoctest
+1 + 2
+
+# output
+
+3
+```
+\"\"\"
+foo() = 1
+end # module
+"""
```

Comment:
Having a test with CRLFs wouldn't hurt. We could also keep these in separate files and just `cp` them over to `tempdir` to avoid problems with differing line endings in a single file (although a simple `replace(..., '\n' => "\r\n")` works as well I guess).